### PR TITLE
Add examples and formal specifications

### LINKED ISSUE #32: Add more formal specification
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
Currently only a tiny spec for the txt format exists in the documentation.

**What you expected to happen**:
Full formal spec for txt format and the various types.

### PR DESCRIPTION
**Pull request by stp-ip**
**What this PR does / why we need it**:
Add more examples and a formal specification

**Which issue this PR fixes** *(optional - uncomment and add issue)*:
fixes #32 #

**Special notes for your reviewer**:
Still needs specifications for each type (examples already done)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -16,32 +16,60 @@ limitations under the License.
 
 For configuration examples for the caddy plugin look at our [examples section](/examples/README.md#configuration)
 
-## Specification
-### General
-* The TXT record must consist of multiple sections, delimited by semicolons(;)
+# Specification
+*implementation does not yet implement the full specification yet*
```
Comment (alarsyo) [inline docs/README.md]: Double yet :)

File: examples/README.md [hunk 2]
```diff
@@ -60,12 +60,34 @@ txtdirect {
...
+{path_escaped} 	Query-escaped variant of {path}
+{port} 	The client's port
+{query} 	The query string portion of the URL, without leading "?"
+{query_escaped} 	The query-escaped variant of {query}
+{?key} 	The value of the "key" argument from the query string
+{remote} 	The client's IP address
+{scheme} 	The protocol/scheme used (usually http or https)
+{uri} 	The request URI (includes path, query string, and fragment)
+{uri_escaped} 	The query-escaped variant of {uri}
+```
```
Comment (alarsyo) [inline examples/README.md]: Weird indentation/alignment in code block ?

Comment (stp-ip) [reply examples/README.md]: Will move it out of the code block.

**Comment by alarsyo** [comment]
Heads up: still a problem with the formatting of the placeholders section when rendered in Markdown

**Pull request merged**